File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int correct_k = k % nums.size();
        reverse(nums.begin(), nums.end());
        reverse(nums.begin(), nums.begin() + correct_k);
        reverse(nums.begin() + correct_k, nums.end());
    }
};

File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {
    let reverse = function (i, j) {
        while (i < j) {
            let temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
            i++
            j--
        }
    }
    let size = nums.length;
    k = k % size;
    if (k === 0) return ;
    reverse(0, size - 1);
    reverse(0, k - 1);
    reverse(k, size - 1);

};